packages/router/src/grpc/custody/authorize.ts

```diff
@@ -18,7 +18,11 @@ import { viewTransactionPlan } from './view-transaction-plan';
 const getMetadataByAssetId = async (ctx: HandlerContext) => {
   const services = ctx.values.get(servicesCtx);
   const walletServices = await services.getWalletServices();
-  const assetsMetadata = await walletServices.indexedDb.getAllAssetsMetadata();
+
+  const assetsMetadata: Metadata[] = [];
+  for await (const metadata of walletServices.indexedDb.iterateAssetsMetadata()) {
+    assetsMetadata.push(metadata);
+  }
```

Comment:
Not for this PR, but we should refactor `viewTransactionPlan()` to accept a parameter that allows for retrieving individual metadata async.

Reply:
```typescript
/**
 * @todo As more asset types get used, the amount of asset metadata we store
 * will grow. Loading all the asset metadata into memory for the purpose of
 * compiling a transaction view may not be sustainable in the long term.
 * Eventually, we may want to scan through the transaction plan, extract all the
 * asset IDs in it, and then query just those from IndexedDB instead of grabbing
 * all of them.
 */
``` 
Looks like there's already a todo for that